lexer/lexer.go

```diff
@@ -105,11 +213,26 @@ func (l *Lexer) getSubshellString() (Node, error) {
 	}
 
 	prevState := l.state
-	l.state = Normal
 	defer func() {
 		l.state = prevState
 	}()
 
+	if l.Input[l.consumed] == '(' {
+		nodes, err := l.getArithmetic()
```

Comment:
Да
> the shell shall first determine whether it can parse the expansion as an arithmetic expansion and shall only parse the expansion as a command substitution if it determines that it cannot parse the expansion as an arithmetic expansion. The shell need not evaluate nested expansions when performing this determination.